### ISSUE TITLE
Request for adding support for installing only runtime targets

### ISSUE THREAD

**Issue by kordejong** [STEllAR-GROUP/hpx#6188]
I maintain the HPX Conda package (https://anaconda.org/conda-forge/hpx). Conda packages get built for various platforms (https://github.com/conda-forge/hpx-feedstock). After being built, they are available for people to install in their Conda environment, for example to be able to build their own software that depends on HPX and which provides a Python API.

To be able to **build a Conda package** that depends on HPX, it is necessary that there is a HPX Conda package containing all the HPX headers, libraries, CMake scripts, etc.

To be able to **use a Conda package** that depends on HPX, it is useful that there is an HPX Conda package containing only the shared libraries and likely some other files (e.g. `hpxrun.py` is useful). This makes the package smaller. These users don't need the static libraries, header files, etc.

I am submitting this issue because I think it is currently not possible to install only the runtime targets of HPX. If this is indeed the case, is support for installing only runtime targets be something that the HPX team is willing to add?

Once I can create seperate development and runtime packages I will update the HPX Conda packaging logic to create a `hpx-dev` and `hpx` package.

**Comment by hkaiser** [comment]
@kordejong, I believe adding `-DHPX_WITH_EXAMPLES=OFF -DHPX_WITH_TESTS=OFF -DHPX_WITH_TOOLS=OFF` to the cmake command line should do what you need. Could you try, please?

**Comment by kordejong** [comment]
Hi @hkaiser , thanks for your suggestion. I am getting close now, but I am in a somewhat confusing corner of CMake behaviour. By default, all install targets are installed "unless marked with `EXCLUDE_FROM_ALL`". I used this command: `cmake --install . --prefix ~/tmp/hpx-dev` So, with your options, all targets, including all header files are installed (2640 files). This is great for a development package `hpx-dev`.
In case no `COMPONENT` option is passed to CMake's `INSTALL` command, a default component named `Unspecified` is used. When explicitly passing that name to the install command, for some reason fewer files get installed. `cmake --install . --prefix ~/tmp/hpx-runtime --component Unspecified` results in 112 files being installed. These are mostly static libraries, and shared libraries and CMake scripts. I think for a runtime HPX package, only shared libraries and the HPX tools would be needed. I don't think the static libraries and CMake scripts should be installed, but maybe I am missing something.
So, it seems support for a lean runtime installation is almost present. Maybe the HPX CMake scripts can explicitly pass `COMPONENT=Runtime` to install commands with which only those parts of HPX needed at runtime are installed (shared libraries, the tools). That would shrink the package to contain 11 shared libraries and the tools.
BTW, I used HPX-1.9.0-rc1.
